client/state/jetpack/credentials/reducer.js

```diff
@@ -63,6 +66,21 @@ export const updateRequestStatus = keyedReducer( 'siteId', ( state, { type } ) =
 	return state;
 } );
 
+export const testRequestStatus = keyedReducer( 'siteId', ( state, { type, role, testResult } ) => {
```

Comment:
I'd suggest adding some tests to cover the reducer functionality that we're introducing here.

Reply:
I will be on it.